server/src/test/java/io/crate/execution/ddl/tables/AlterTableOperationTest.java

```diff
@@ -58,6 +65,32 @@ public void testValidateReadOnlyForResizeOperation() {
         AlterTableOperation.validateReadOnlyIndexForResize(indexMetadata);
     }
 
+    @Test
+    public void testValidateSettingForPublishedTables() {
+        Settings settings = Settings.builder()
+            .put(SETTING_NUMBER_OF_SHARDS, 5L)
+            .build();
+
+        RelationName t1 = new RelationName("doc", "t1");
+        var oneTablePublished = Map.of("pub1", new Publication("owner", false, List.of(t1)));
+
+        assertThrowsMatches(() -> AlterTableOperation.validateSettingsForPublishedTables(t1,
+                                                                                         settings,
+                                                                                         oneTablePublished,
+                                                                                         IndexScopedSettings.DEFAULT_SCOPED_SETTINGS),
+                            IllegalArgumentException.class,
+                            "Setting [index.number_of_shards] cannot be applied to table 'doc.t1' because it is included in a logical replication publication 'pub1'");
+
+        var allTablesPublished = Map.of("pub1", new Publication("owner", true, List.of()));
+
+        assertThrowsMatches(() -> AlterTableOperation.validateSettingsForPublishedTables(t1,
+                                                                                         settings,
+                                                                                         allTablesPublished,
+                                                                                         IndexScopedSettings.DEFAULT_SCOPED_SETTINGS),
+                            IllegalArgumentException.class,
+                            "Setting [index.number_of_shards] cannot be applied to table 'doc.t1' because it is included in a logical replication publication 'pub1'");
```

Comment:
nitpick: putting the first arg of the calls into a newline would remove quite some indents and such increase readability.